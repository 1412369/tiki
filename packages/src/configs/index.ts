export * from './styled';
export * from './message';
export const API_OFFSET: number = 10;
export const APP_HELMET = {
  title:
    'Vé hòa nhạc, vé hội thảo, vé sự kiện, vé thể thao ở Việt Nam | TicketBox',
  image: 'https://static.ticketbox.vn/images-v2/TKB-FBthumb.jpg',
  description:
    'TicketBox là hệ thống phân phối vé và quản lý sự kiện trực tuyến, giúp bạn bán và mua vé các sự kiện thật dễ dàng.',
  url: 'https://ticketbox.vn',
};
export const COLORS = {
  vip: {
    default: '#008e00',
  },
  deluxe: {
    default: '#f76f6f',
  },
  standard: {
    default: '#464646',
  },
  selected: '#ea8383',
  hover: '#b9b9b9',
  isPicked: '#8e8e8e',
};

export const currencyFormat = new Intl.NumberFormat('vi-VN', {
  style: 'currency',
  currency: 'VND',
});
export const countTicketByStatus = (tickets: any[], status: string): number => {
  let count = 0;
  for (let i = 0; i < tickets.length; i++) {
    for (let j = 0; j < tickets[i].length; j++) {
      if (tickets[i][j][status]) {
        count += 1;
      }
    }
  }
  return count;
};
export const calculatePrice = (selected: any, meta: any): any => {
  return Object.keys(selected.tickets).reduce((results, item) => {
    const ticket = selected.tickets[item];
    if (ticket) {
      return results + meta[ticket.type].price;
    }
    return results;
  }, 0);
};
