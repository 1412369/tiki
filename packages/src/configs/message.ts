export const MSG_EMPTY_RESULT = 'Không tìm thấy kết quả';
