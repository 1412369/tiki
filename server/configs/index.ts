export const cacheTime = 3600000 * 24 * 30;
export const generateEmptyMap = ({
  row,
  col,
  vip,
  deluxe,
  picked,
}: {
  row: number;
  col: number;
  vip: number[];
  deluxe: number[];
  picked: number[];
}) => {
  const FROM = 0,
    TO = 1;
  return [...new Array(row)].map((_, rowIndex) =>
    [...new Array(col)].map((_, colIndex) => {
      const flatIndex: number = rowIndex * col + colIndex;
      if (flatIndex >= vip[FROM] && flatIndex <= vip[TO]) {
        return {
          type: 'vip',
          isPicked: picked.includes(flatIndex),
        };
      }
      if (flatIndex >= deluxe[FROM] && flatIndex <= deluxe[TO]) {
        return {
          type: 'deluxe',
          isPicked: picked.includes(flatIndex),
        };
      }
      return {
        type: 'standard',
        isPicked: picked.includes(flatIndex),
      };
    })
  );
};
