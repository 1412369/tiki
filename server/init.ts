import LRU from 'lru-cache';
const options = {
  max: 50,
  maxAge: 1000 * 60 * 60,
};
global.lruCache = new LRU(options);
