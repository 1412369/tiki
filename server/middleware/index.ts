export { default as webpackMiddleware } from './webpack';
