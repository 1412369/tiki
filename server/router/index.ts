export { default as AppRouter } from './app/app.router';
