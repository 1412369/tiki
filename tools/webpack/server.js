const path = require('path');
var nodeExternals = require('webpack-node-externals');
const { webpackCore } = require('./common');

module.exports = webpackCore({
  entry: {
    server: `./server/index.ts`,
  },
  output: {
    path: `${process.cwd()}/dist/server`,
    filename: `[name].js`,
  },
  mode: 'development',
  devtool: 'source-map',
  externals: [nodeExternals()],
  target: 'node',
});
