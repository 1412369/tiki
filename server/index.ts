import './init';

import * as bodyParser from 'body-parser';
import * as compression from 'compression';
import * as cookieParser from 'cookie-parser';
import * as cors from 'cors';
import * as express from 'express';

import * as helmet from 'helmet';

import { AppRouter } from './router';
import { cacheTime } from './configs';
import { webpackMiddleware } from './middleware';

declare var ENVS: any;
const app: express.Application = express();
//============== init middleware ==============
(() => {
  app.use(cookieParser());
  app.use(cors());
  app.use(express.static('public', { maxAge: cacheTime }));
  app.use(express.static('dist'));
  app.use(bodyParser.json());
  if (ENVS.NODE_ENV === 'production') {
    app.use(compression());
    app.use(helmet());
  } else {
    webpackMiddleware(app);
  }
})();

//============== init router ==============
(() => {
  app.use(AppRouter);
})();

const PORT: number = parseInt(ENVS.PORT, 10);
app.listen(PORT, () => {
  console.log('Running on http://localhost:' + PORT);
});
