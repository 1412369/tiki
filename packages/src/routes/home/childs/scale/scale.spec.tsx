import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { countTicketByStatus } from '../../../../configs';
import { generateEmptyMap } from '../../../../../../server/configs';
import Scale from './scale';
let picked = [1, 2, 3, 4, 7, 8, 20, 21, 22, 23];
let tickets: any = generateEmptyMap({
  row: 6,
  col: 6,
  vip: [10, 20],
  deluxe: [21, 29],
  picked,
});
const meta = {
  total: 36,
  types: ['standard', 'vip', 'deluxe'],
  standard: { price: 10000 },
  vip: { price: 12000 },
  deluxe: { price: 15000 },
};
const selected = {
  tickets: {
    0_0: { type: 'vip' },
    0_1: { type: 'vip' },
    0_2: { type: 'vip' },
  },
};
describe('countTicketByStatus', function() {
  it('count::isPicked', function() {
    expect(countTicketByStatus(tickets, 'isPicked')).toEqual(picked.length);
  });

  it('count::non status ticket', function() {
    expect(countTicketByStatus(tickets, 'test')).toEqual(0);
  });
});
describe('snapshot Scale', function() {
  it('Should render Scale without crashing', function() {
    const wrapper = renderer.create(
      <Scale
        scale={false}
        setScale={() => {}}
        selected={selected}
        meta={meta}
        tickets={tickets}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
