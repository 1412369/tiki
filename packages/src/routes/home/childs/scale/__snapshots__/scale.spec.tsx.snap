// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Scale Should render Scale without crashing 1`] = `
<div
  className="sc-bdVaJa knOAqe"
>
  <button
    onClick={[Function]}
  >
    zoom 
    +
  </button>
  <ul>
    <li>
      Đã chọn:
      <div
        className="sc-bxivhb eDmdrw"
      >
        10
      </div>
    </li>
    <li>
      Đang chọn:
       
      <div
        className="sc-bxivhb bDiUoi"
      />
    </li>
    <li>
      Trống:
       
      <div
        className="sc-bxivhb cMurYG"
        style={
          Object {
            "color": "black",
          }
        }
      >
        26
      </div>
    </li>
  </ul>
</div>
`;
